src/libsync/discoveryphase.cpp

```diff
@@ -374,12 +374,26 @@ void DiscoverySingleDirectoryJob::directoryListingIteratedSlot(QString file, con
             file = file.remove(0, 1);
         }
 
-
-        std::unique_ptr<csync_file_stat_t> file_stat(propertyMapToFileStat(map));
+        std::unique_ptr<csync_file_stat_t> file_stat(new csync_file_stat_t);
         file_stat->path = file.toUtf8();
-        if (file_stat->etag.isEmpty()) {
-            qCCritical(lcDiscovery) << "etag of" << file_stat->path << "is" << file_stat->etag << "This must not happen.";
+        file_stat->size = -1;
+        file_stat->modtime = -1;
+        propertyMapToFileStat(map, file_stat.get());
+        if (file_stat->type == ItemTypeDirectory)
+            file_stat->size = 0;
+        if (file_stat->type == ItemTypeSkip
+            || file_stat->size == -1
+            || file_stat->modtime == -1
+            || file_stat->remotePerm.isNull()
+            || file_stat->etag.isEmpty()
+            || file_stat->file_id.isEmpty()) {
+            _error = tr("The server file discovery reply is missing data.");
+            qCCritical(lcDiscovery)
```

Comment:
i think qCWarning should be enough compared to the other reason we use qcWarning